### ISSUE TITLE
it doesn't scroll page, if item to be highleted is somewhere down.

### ISSUE THREAD

**Issue by moghya** [kamranahmedse/driver.js#2]

**Comment by moghya** [comment]
@kamranahmedse I want to contribute how do I approach?

**Comment by kamranahmedse** [comment]
Could you please the details about your browser and platform? It seems to be fine for me.

And for the contribution, just clone it `yarn install`, `yarn start`, make your changes, `yarn build` and open PR :)

**Comment by moghya** [comment]
@kamranahmedse Firefox Quantum 57.0.1 (64-bit) on Ubuntu 16.04 (64-bit)

**Comment by kamranahmedse** [comment]
Unfortunately, I don't have the resources at my disposal. Could you possibly try to debug [this section](https://github.com/kamranahmedse/driver.js/blob/master/src/core/element.js#L77) if/when you have time till I get the time/resources to try it for myself?

**Comment by moghya** [comment]
@kamranahmedse apparently 
`
TypeError: 'block' member of ScrollIntoViewOptions 'center' is not a valid value for enumeration ScrollLogicalPosition.`

this is the error on console caused on calling `bringInView()`

**Comment by kamranahmedse** [comment]
Should be fine now, just deployed the demo page. Could you please test and close if fixed :)

**Comment by moghya** [comment]
It works now. :+1:

**Comment by kamranahmedse** [comment]
Thanks

**Comment by kapooramit** [comment]
It is working if the desired scroll is downward. However, if the element is at the top and you start the tour while at the bottom of a long page, the tour just brings the element in the view without scrolling to it and shows the overlays and popups. The elements look distorted then as they are not where they are supposed to be. How to fix this? Something to do with positioning?

**Comment by kapooramit** [comment]
@moghya , @kamranahmedse : Can you suggest?

**Comment by moghya** [comment]
@kapooramit can you please provide browser and system details? If possible can you create a Minimal Example to look into.

**Comment by kapooramit** [comment]
Sure. Firefxo 68.0.1 (64-bit), Windows 10 64-bit Enterprise. Code being developed in K15t Scroll Viewport.

**Comment by moghya** [comment]
@kamranahmedse I think I made mistake [here](https://github.com/kamranahmedse/driver.js/blob/df9c1b535e105290e5b3c3813a2b1b35658dd0ba/src/core/element.js#L77) whilie fixing this. 

I made an assumption that element to bring in view is always below current scroll position, and hence adding this.window.pageYOffset to the elementRect.top.

  ```
/**
   * Manually scrolls to the position of element if `scrollIntoView` fails
   * @private
   */
  scrollManually() {
    const elementRect = this.node.getBoundingClientRect();
	// this line here is incorrect.
    const absoluteElementTop = elementRect.top + this.window.pageYOffset;
    const middle = absoluteElementTop - (this.window.innerHeight / 2);

    this.window.scrollTo(0, middle);
  }
```
working on a fix,

**Comment by moghya** [comment]
@kapooramit well after a bit of trying some examples, I think your problem is not because of some bug in code. 

Basically you want 'smooth' scroll, for that you can set `behavior: 'smooth'` in your `scrollIntoViewOptions`. 

While creatting driver object in `options` add `scrollIntoViewOptions` with `behavior: 'smooth'` key-value pair. 

@kamranahmedse  I hope this is what he wanted.

**Comment by kapooramit** [comment]
@moghya : Interestingly, ```behavior:'smooth'``` made the movement smooth but did not solve my issue. But with your guidance to use scrollIntoViewOptions, I tried , ```block:'top'``` and that solved the issue. Now, no strange appearence of the element that belongs to the top of the page floating in the middle of nowhere at the bottom part of page - instead, now the scroll actually moves to the top and highlights the element as required. 
Strangely with this, ```behavior:'smooth'``` does not work and it works as if it is set to (default) ```behavior:'instant'```. So I have removed ```behavior:'smooth'``` because it is not doing anything.

**Comment by kapooramit** [comment]
Thanks @moghya  and thank you @kamranahmedse .

**Comment by maciejklocek** [comment]
Hello,
I've encountered a similar issue as explained above - when the element is at the top of the page the tour does indeed show overlay and popup however it does not scroll up to that element.

I've tried changing the scroll options as @moghya and @kapooramit suggested but behavior:'smooth' does nothing to solve the issue (just like @kapooramit said) and when trying to change block property to 'top' I get an error "Type '"top"' is not assignable to type 'ScrollLogicalPosition'." and scrolling behaves in an odd way - i.e. it seems to scroll properly to the element that is outside the window visibility BUT when it jumps to that element then everything that wasn't visible in initial window visibility is not displayed at all.

_web browser: Chrome 90.0.4430.212 (64-bit)_

**Is there any change the issue would be looked into again?**
